src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -110,50 +117,75 @@ void Run (DirectoryAssemblyResolver res)
 			}
 
 			// Put every assembly we'll need in the resolver
+			bool hasExportReference = false;
+			bool haveMonoAndroid = false;
+			var all_assemblies = new Dictionary<string, bool> (StringComparer.OrdinalIgnoreCase);
+			var user_assemblies = new List<string> ();
 			foreach (var assembly in ResolvedAssemblies) {
-				if (bool.TryParse (assembly.GetMetadata (AndroidSkipJavaStubGeneration), out bool value) && value) {
+				if (String.Compare (assembly.GetMetadata (AndroidSkipJavaStubGeneration), "true", StringComparison.OrdinalIgnoreCase) == 0) {
```

Comment:
I think `bool.TryParse` is OK, it does the same `string.Compare`: 
https://github.com/mono/mono/blob/d44a5d3a0bf24970a8cb03ddc1f9c94dcfa57949/mcs/class/referencesource/mscorlib/system/boolean.cs#L173-L203